packages/expo-modules-autolinking/scripts/ios/autolinking_manager.rb

```diff
@@ -158,5 +166,14 @@ class AutolinkingManager
       ])
     end
 
+    private def use_modular_headers_for_dependencies(dependencies)
+      dependencies.each { |dependency|
+        unless @target_definition.build_pod_as_module?(dependency.name)
+          # This is an equivalent to setting `:modular_headers => true` for the specific dependency.
+          @target_definition.set_use_modular_headers_for_pod(dependency.name, true)
```

Comment:
could we add some log for the dependency that we override the module headers setting?

Reply:
I thought about this, but it may output some dependencies that already define modules. We can't really filter out them from `use_expo_modules!` function since we don't yet know their xcconfig. So for example, in bare-expo, even without `expo-image` it would log `ExpoModulesCore`, `ReactCommon/turbomodule/core` and `EXImageLoader` that all already define modules.

Reply:
I just added the log above